akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -751,25 +772,50 @@ private[akka] class ShardRegion(
   }
 
   def replyToRegionStatsQuery(ref: ActorRef): Unit = {
-    askAllShards[Shard.ShardStats](Shard.GetShardStats)
-      .map { shardStats =>
-        ShardRegionStats(shardStats.map {
-          case (shardId, stats) => (shardId, stats.entityCount)
-        }.toMap)
+    queryShards[ShardStats](shards, Shard.GetShardStats)
+      .map {
+        case (timeouts, shardStats) =>
+          val qm = ShardingQueries.Metadata(shards.size, timeouts, shardStats)
+          if (qm.unresponsiveNonEmpty) log.warning(qm.toString)
+
+          // TODO add ShardRegionStats field for timeouts (proto): ShardRegion.ShardRegionStats(stats, timeouts.toSet)
+          // Until then
+          val stats = shardStats.map(stats => (stats.shardId, stats.entityCount)).toMap ++ timeouts.map((_, 0))
+          ShardRegionStats(stats)
```

Comment:
Ok, so the PR isn't quite ready, because we don't actually use the info (but still break source and bincomp with the marker trait renaming?)

Reply:
I guess to be ready I'd need to go all in here and make a proto change and attach to parent ticket instead. But I thought better to do in two tickets so the idea/change is not so large.

Reply:
I would have liked it better if it was all in this PR since the whole motivation of the changes, to get the info back to the user, now is elsewhere.

Reply:
This last commit simplifies, based on feedback so I think the two tickets work now.